src/components/virtual-scroll/virtual-scroll.ts

```diff
@@ -420,8 +441,9 @@ export class VirtualScroll implements DoCheck, AfterContentInit, OnDestroy {
 
     let needClean = false;
     if (changes) {
+      var lastRecord = this.lastRecord() + 1;
       changes.forEachOperation((item, _, cindex) => {
-        if (item.previousIndex != null || (cindex < this._recordSize)) {
+        if (item.previousIndex != null || (cindex < lastRecord)) {
```

Comment:
The if here will only filter out adds after last record.
I tried something like:
```
        // add new record after current position
        if (item.previousIndex === null && cindex > lastRecord) {
          console.debug('adding record after current position, no recalculation needed');
          return;
        }

        // remove record after current position
        if (item.previousIndex > lastRecord && cindex == null) {
          console.debug('removing record after current position, no recalculation needed');
          return;
        }

        needClean = true;
```

to add more explicit checks for filtered out cases (we can probably consolidate them later on)
This now allows 3 cases to not trigger cleaning (the first two obvious, the third not so much). 
1.  Adding a record after lastRecord
2. Removing a record after lastRecord
3. Changing a record after lastRecord.

When changing the reference of a record, changes is populated with two entries, one for the removed record and on for the added record. So forEachOperation will be triggered twice, with half the info each time.

logs after triggering a random record changed that happened on position 179:
>virtual-scroll.ts:451 virtual-scroll, prevIndex: 179 cindex: null lastRecord: 33
>virtual-scroll.ts:460 removing record after current position, no recalculation needed
>virtual-scroll.ts:451 virtual-scroll, prevIndex: null cindex: 179 lastRecord: 33
>virtual-scroll.ts:454 adding record after current position, no recalculation needed